### ISSUE TITLE
getTicketVersion() fails for create

### ISSUE THREAD

**Issue by mpedrummer** [jakoch/PHPTracRPC#22]
I'm not entirely sure the "right" way to fix this one - the API is a real bear to figure out, since all the parameters are just documented as "struct attributes" without documenting the name/value pairs that the struct should contain!

Anyway, for 'create' and 'update' the $attr array is not passed in appropriately - the $params array is never updated, so it remains at the default value, which is an empty array.

**Comment by mpedrummer** [comment]
I'm able to get it to create the milestone, but can't figure out how to get it to mark the release date properly.

**Comment by jakoch** [comment]
Hey Tim!

Please give the [`function getTicketUpdate($action = 'create', $id = '', $data = array())`](https://github.com/jakoch/PHPTracRPC/blob/27528306545f48d65ae9f448c288c1ed9a20f3d0/lib/TracRPC.php#L418) a try. Maybe its closer to what you are trying to do.

For the name/value pairs the DB schema might be of help: https://trac.edgewall.org/wiki/TracDev/DatabaseSchema

> I'm able to get it to create the milestone, but can't figure out how to get it to mark the release date properly.

Thats `ticket.milestone.update`, right? So, its `public function getTicketMilestone($action = 'get_all', $name = '', $attr = array())` with `action` `'update'`.  You could try to reuse the attributes from the `'get'` action on the `'update'` call.

If this doesn't work out, i need to setup a new Trac endpoint for testing `*.create|update|delete` calls.

**Comment by mpedrummer** [comment]
No, I'm just a twit and misspoke - I meant version.  I'm automating our build process to create the new versions in Trac.

That db schema looks like an excellent resource - I'll give it a go tomorrow and see if it correlates to what I'm doing.
